lib/ansible/plugins/action/copy.py

```diff
@@ -407,7 +407,7 @@ def run(self, tmp=None, task_vars=None):
         local_follow = boolean(self._task.args.get('local_follow', True), strict=False)
 
         result['failed'] = True
-        if (source is None and content is None) or dest is None:
+        if ((not source or source is None) and content is None) or dest is None:
```

Comment:
I would split it out and have more specific error messages:
```python
    if not source and content is None:
        result['msg'] = "src (or content) is required"
    elif dest is not None:
        result['msg'] = "dest is required"
    elif source and content is not None:
        result['msg'] = "src and content are mutually exclusive"
```
Can dest be an empty string ? (current script directory ?)